src/screens/MainApp/DiagnoseRequest/ImageReview/index.js

```diff
@@ -0,0 +1,136 @@
+    setImages([...images, filterDuplicates[0]])
+  }
+
+  const FlatListItem = ({ item }) => {
+    const itemIndex = images.indexOf(item)
+    const lastIndex = images.length - 1
+    const selected = images[activeIndex] === item
+    if (itemIndex === lastIndex && images.length < MAX_IMAGES) {
+      return (
+        <>
+          <FlatListImage selected={selected} item={item} onPress={() => setCarouselItem(images.indexOf(item))} />
+          <FlatListImage item={ADD_MORE_IMAGES_URI} onPress={getMorePhotos} />
+        </>
+      )
+    }
+    return (
+      <FlatListImage selected={selected} item={item} onPress={() => setCarouselItem(images.indexOf(item))} />
+    )
+  }
+
+  useEffect(() => {
+    if (images.length < MIN_IMAGES) {
+      setSubmitError(true)
+    } else {
+      setSubmitError(false)
+    }
+
+    const deleteImage = () => {
+      if (images.length === 1) {
+        setImages([])
+        NavigationService.navigate('NoPhotoDisclaimer')
+      } else {
+        const wantedImages = images.filter((val, index) => index !== activeIndex)
+
+        setImages(wantedImages)
+
+        if (activeIndex === 0) {
+          setFlatListItem(0)
+          setCarouselItem(0)
+        }
+
+        if (activeIndex > 0) {
+          const nextIndex = activeIndex - 1
+          setFlatListItem(nextIndex)
+          setCarouselItem(nextIndex)
+        }
+      }
+    }
+
+    const changeImage = () => {
+      const changedImageArray = images.slice()
+      changedImageArray[activeIndex] = newImages[activeIndex]
+      setImages(changedImageArray)
+    }
+
+    props.navigation.setParams({ deleteImage })
+    props.navigation.setParams({ changeImage })
+  }, [activeIndex, images.length]) // this is needed to bind to the static header on changes
+
+  return (
+    <View>
+      <View>
+        <Carousel
+          ref={carouselRef}
+          data={images}
+          renderItem={({ item }) => <CarouselImage item={item} />}
+          sliderWidth={CAROUSEL_SLIDER_WIDTH}
+          itemWidth={ITEM_WIDTH}
+          itemHeight={ITEM_HEIGHT}
+          onSnapToItem={(index) => setFlatListItem(index)}
+        />
+        <Pagination
+          dotsLength={images.length}
+          activeDotIndex={activeIndex}
+          carouselRef={carouselRef}
+          dotStyle={styles.paginationDot}
+          inactiveDotStyle={styles.inactivePaginationDot}
+        />
+      </View>
+      <FlatList
+        ref={flatListRef}
+        horizontal
+        data={images}
+        showsHorizontalScrollIndicator={false}
+        renderItem={({ item }) => <FlatListItem item={item} />}
+        keyExtractor={item => item}
+      />
+      <ConfirmButton errorState={submitError} onConfirm={confirmRequest} />
+    </View>
+  )
+}
+
+ImageVisualization.navigationOptions = ({ navigation }) => ({
+  headerRight: () => (
+    <>
+      <Button variant='alpha' onPress={navigation.getParam('changeImage')}><Icon name='redo' size={18} color='black' /></Button>
```

Comment:
You're closing the component twice, is that for a special reason?

Reply:
I will write this different. There is actually an Icon component which is closed before the Button tag.